app/src/main/java/com/example/series_collector/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.series_collector.ui.home
+
+import android.content.Context
+import androidx.lifecycle.*
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
+import com.example.series_collector.data.Series
+import com.example.series_collector.data.repository.SeriesRepository
+import com.example.series_collector.utils.workers.SeriesInitWorker.Companion.WORK_TAG
+import com.example.series_collector.utils.workers.Workers
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class HomeViewModel @Inject constructor(
+    private val seriesRepository: SeriesRepository,
+    private val workers: Workers,
+    @ApplicationContext appContext: Context
+) : ViewModel() {
+
+    val outputInitWorkInfos: LiveData<List<WorkInfo>> = WorkManager.getInstance(appContext)
+        .getWorkInfosByTagLiveData(WORK_TAG)
+
+    init {
+        viewModelScope.launch {
+            if (seriesRepository.isEmpty()) {
+                workers.buildInitWorker(appContext)
+            } else {
+                insertUpdatedSeries()
+            }
+        }
```

Comment:
```suggestion
        viewModelScope.launch {
            if (seriesRepository.isEmpty()) {
                workers.buildInitWorker(appContext)
            } else {
                insertUpdatedSeries()
            }
        }
```
최초에만 worker를 통해 data fetch를 진행하고 업데이트는 viewmodel 내부에서 data fetch를 진행하는걸로보이는데
이렇게 구현하신 이유가있으실까요?
둘다 worker에서 처리하도록 하는게 적절해보입니다 :)